enterprise/coderd/prebuilds/reconcile_test.go

```diff
@@ -1034,15 +1034,18 @@ func TestHardLimitedPresetShouldNotBlockDeletion(t *testing.T) {
 			require.Equal(t, database.WorkspaceTransitionDelete, workspaceBuilds[0].Transition)
 			require.Equal(t, database.WorkspaceTransitionStart, workspaceBuilds[1].Transition)
 
-			// Metric is deleted after preset became outdated.
+			// The metric is still set to 1, even though the preset has become outdated.
+			// This happens because the old value hasn't been overwritten by a newer preset yet.
```

Comment:
nit: In `trackHardLimitedStatus`, when a preset is no longer hard limited, instead of deleting the entry immediately from the `isPresetHardLimited` map, we just set it to `false`.
Then, in the `Collect` function, we report the metric with value `0` one last time, and after that we delete the entry from the map, so it's no longer reported in future scrapes.
This ensures that while Prometheus still hasn’t cleaned up the time series, it holds the correct last value (`0`) rather than incorrectly keeping `1`.